miner/worker.go

```diff
@@ -1050,7 +1050,8 @@ func (w *worker) prepareWork(genParams *generateParams) (*environment, error) {
 // fillTransactions retrieves the pending transactions from the txpool and fills them
 // into the given sealing block. The transaction selection and ordering strategy can
 // be customized with the plugin in the future.
-func (w *worker) fillTransactions(interrupt *int32, env *environment) {
+// Returns whether block should be discarded.
+func (w *worker) fillTransactions(interrupt *int32, env *environment) bool {
```

Comment:
It is very unorthodox to use a bool as a failure signal, and even moreso to mean `true` == failed and `false` as success (i.e. `if fillTransactions() { then failed }`).

My recommendation is to swap the return for an `error` and then you can more meaningfully check it and either propagate it upwards, or act on it immadiately.

Reply:
I agree, this should be the case for the whole stack.
Initially I didn't want to refactor the whole thing, but since you seem to also prefer it this way I'll go ahead.

Reply:
For reference, this was keeping consistency with `commitTransactions` - refactored both.